File: mcdowell/php/ch1-08.php
<?php

function isRotation($s1, $s2) {

    if (!is_string($s1) || !is_string($s2)) {
        throw new Exception("Inputs must be strings.");
    }

    if (!$s1 || !$s2 || strlen($s1) != strlen($s2)) return false;

    return strpos($s1.$s1, $s2) !== false;
}


var_dump(isRotation('abcd', 'cdab'));
var_dump(isRotation('abcasdfsdfd', 'cdab'));

File: mcdowell/python/ch1-08.py
def isRotation(s1, s2):

    if not s1 or not s2 or len(s1) != len(s2):
        return False

    return (s1+s1).find(s2) != -1



print isRotation('abcd', 'cdab')
print isRotation('abcd', 'aaaa')
print isRotation('abcasdfsdfd', 'cdab')
